### PR TITLE
I have a hatred for warnings

### PR DESCRIPTION
**Pull request by adamemerson**
I wish all the compiler warnings in the repository had just one throat with my fingers clamped around it.

### CODE REVIEW

File: src/test/librados/c_write_operations.cc [hunk 1]
```diff
@@ -211,7 +211,8 @@ TEST(LibRadosCWriteOps, WriteSame) {
   rados_write_op_writesame(op, "four", 4, 4 * 4, 0);
   ASSERT_EQ(0, rados_write_op_operate(op, ioctx, "test", NULL, 0));
   char hi[4 * 4];
-  ASSERT_EQ(sizeof(hi), rados_read(ioctx, "test", hi, sizeof(hi), 0));
+  ASSERT_EQ(sizeof(hi),
+	    (std::size_t) rados_read(ioctx, "test", hi, sizeof(hi), 0));
```
Comment (gregsfortytwo) [inline src/test/librados/c_write_operations.cc]: We're trying to switch to proper C++ casts instead of C ones.

static_cast<std::size_t>( /*stuff */ )
etc

**Comment by gregsfortytwo** [comment]
I support your drive to exterminate warnings!

File: src/test/librados/c_write_operations.cc (hunk 1 above)
Comment (adamemerson) [inline src/test/librados/c_write_operations.cc]: Fixed!

**Comment by wjwithagen** [comment]
On 3-5-2016 19:00, Adam C. Emerson wrote:
> I wish all the compiler warnings in the repository had just one throat
> with my fingers clamped around it.
> 

I can second that.
Clang is throwing quite a fit on some parts of the code.

Be it only the mix between class en struct definitions.
Got a stack of them pilled up in my local wip, but do not want to commit
these as I'm still working on them.

--WjW

**Pull request merged**